internal/backend/crypto/gpg/cli/encrypt.go

```diff
@@ -25,25 +28,37 @@ func (g *GPG) Encrypt(ctx context.Context, plaintext []byte, recipients []string
 		args = append(args, "--trust-model=always")
 	}
 
+	if len(recipients) == 0 {
+		debug.Log("recipients list is empty!")
+	} else {
+		debug.Log("recipients are %s", recipients)
+	}
+	var badRecipients []string
 	for _, r := range recipients {
 		kl, err := g.listKeys(ctx, "public", r)
 		if err != nil {
 			debug.Log("Failed to check key %s. Adding anyway. %s", err)
 		} else if len(kl.UseableKeys(gpg.IsAlwaysTrust(ctx))) < 1 {
-			out.Printf(ctx, "Not using invalid key %s for encryption. (Check its expiration date or its encryption capabilities.)", r)
+			badRecipients = append(badRecipients, r)
+			errmsg := fmt.Sprintf("Not using invalid key %s for encryption. Check its expiration date, its encryption capabilities and trust.", r)
+			debug.Log(errmsg)
+			out.Printf(ctx, errmsg)
 
 			continue
 		}
+		debug.Log("adding recipient %s", r)
 		args = append(args, "--recipient", r)
 	}
-
 	buf := &bytes.Buffer{}
+	if len(badRecipients) == len(recipients) && len(recipients) != 0 {
```

Comment:
```suggestion
	if len(badRecipients) == len(recipients) {
```
`len(badRecipient)` is necessarily equal to 0 too if `len(recipients)` is 0, meaning this `if` would be true then, why wouldn't we want to take that return then?

I don't realize if there are valid usecases where `len(recipients) == 0` and we want to carry on with the `exec` below?

Reply:
TestEncrypt is a simple (silly?) test that doesn't do much. If it fails because we introduce a new check feel free to update the expecation (i.e. change line 22 to `require.Error(t, err)`).

Reply:
> TestEncrypt is a simple (silly?) test that doesn't do much. If it fails because we introduce a new check feel free to update the expectation (i.e. change line 22 to `require.Error(t, err)`).

OK I did it. It is a bit even more silly now when I don't fully test the encrypt function. It'd be nice to fix that, but I'm not sure if that is possible when `g.binary` is set to `"true"`.